Indicators/RelativeVolume.cs

```diff
@@ -0,0 +1,154 @@
+            : base(name)
+        {
+            _previousDay = default(DateTime);
+            WarmUpPeriod = period;
+        }
+
+        /// <summary>
+        /// Computes the next value for this indicator from the given state.
+        /// </summary>
+        /// <param name="input">The input value to this indicator on this time step</param>
+        /// <returns>A a value for this indicator</returns>
+        protected override decimal ComputeNextValue(TradeBar input)
+        {
+            if (input.Time.Day != _previousDay.Day)
+            {
+
+                CurrentVolume.Clear();
+                _previousDay = input.Time;
+
+                List<DateTime> allDates = new List<DateTime>();
+                allDates = HistoricalVolumes.Keys.ToList();
+                allDates.Sort((ps1, ps2) => DateTime.Compare(ps1, ps2));
+
+                if (IsReady && allDates.Count != WarmUpPeriod)
+                {
+                    var index = 0;
+                    var delta = allDates.Count - WarmUpPeriod;
+                    foreach (DateTime time in allDates)
+                    {
+                        if (index < delta)
+                        {
+                            HistoricalVolumes.Remove(time);
+                        }
+                        index += 1;
+                    }
+                }
+            }
+
+            if (!IsReady)
+            {
+                CurrentVolume.Add(input.Volume);
+                HistoricalVolumes.Add(input.Time, input.Volume);
+                return 0;
+            }
+
+            CurrentVolume.Add(input.Volume);
+            HistoricalVolumes.Add(input.Time, input.Volume);
+
+            var todaysTotal = CurrentVolume.Sum();
+            List<decimal> relativeValues = new List<decimal>();
+            var curDay = -1;
+            decimal curValue = 0;
+            foreach (KeyValuePair<DateTime, decimal> pair in HistoricalVolumes)
+            {
+                if (curDay != pair.Key.Day)
+                {
+                    if (curValue != 0)
+                    {
+                        relativeValues.Add(curValue);
+                    }
+                    curValue = 0;
+                }
+
+                var dayDelta = input.Time - pair.Key;
+                if ((dayDelta.Days >= 1) && (pair.Key.Hour <= input.Time.Hour) && (pair.Key.Minute <= input.Time.Minute) && (input.Time.Second <= pair.Key.Second))
+                {
+                    curValue += pair.Value;
+                }
+                curDay = pair.Key.Day;
+            }
+
+            if (curValue != 0) { relativeValues.Add(curValue); }
+
+            if (relativeValues.Count == 0)
+            {
+                throw new ArgumentException("Need atleast one day of data", nameof(WarmUpPeriod));
+            }
+
+            var relativeAverage = relativeValues.Average();
+            var relativeVolume = todaysTotal / relativeAverage;
```

Comment:
`relativeAverage` could be calculated only when a new day passes?

Reply:
This needs to be calculated every time a new data point comes in. For example, if the Resolution is Hour, period is 14, and its 1:00pm. We need the "realtiveVolumes" from 2 days ago and 1 day ago from open to 1:00pm.

Reply:
> if the Resolution is Hour, period is 14, and its 1:00pm. We need the "realtiveVolumes" from 2 days ago and 1 day ago from open to 1:00pm.

Since this is the `RelativeDailyVolume` indicator I would expect period 14 to mean 14 days? As of current implementation the indicator does not know what resolution the user input. `"realtiveVolumes" from 2 days ago` not sure how '2' days ago is reached here?

> This needs to be calculated every time a new data point comes in.

What if you just kept a SMA instance per each time seen? that would avoid having to calculate anything I believe, it would all be hidden internally by the SMA?